tests/Unit/Http/Validation/Rules/ShirtSizeTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Test\Unit\Http\Validation\Rules;
+
+use Engelsystem\Http\Validation\Rules\ShirtSize;
+use Engelsystem\Test\Unit\ServiceProviderTest;
+
+class ShirtSizeTest extends ServiceProviderTest
+{
+    private ShirtSize $subject;
+
+    public function setUp(): void
+    {
+        $app = $this->createAndSetUpAppWithConfig([]);
+        $app->get('config')->set('tshirt_sizes', $this->loadDefaultShirtSizes());
+        $this->subject = new ShirtSize();
+    }
+
+    /**
+     * @return array<string, array{string, bool}>
+     */
+    public function provideValidateWithDefaultConfigTestData(): array
+    {
+        $data =  [
+            'empty string' => ['', false],
+            'not in list' => ['XXXXXXXL', false],
+        ];
+
+        $defaultShirtKeys = array_keys($this->loadDefaultShirtSizes());
+
+        foreach ($defaultShirtKeys as $defaultShirtKey) {
+            $data[$defaultShirtKey] = [$defaultShirtKey, true];
+        }
+
+        return $data;
+    }
+
+    /**
+     * @covers \Engelsystem\Http\Validation\Rules\ShirtSize::__construct
+     * @dataProvider provideValidateWithDefaultConfigTestData
+     */
+    public function testValidateWithDefaultConfig(mixed $value, bool $expectedValid): void
+    {
+        self::assertSame($expectedValid, $this->subject->validate($value));
+    }
+
+    /**
+     * @return array<string>
+     */
+    private function loadDefaultShirtSizes(): array
+    {
+        $defaultConfig = include __DIR__ . '/../../../../../config/config.default.php';
```

Comment:
Using the default config is not the best idea as users might change it (esp. for tshirts). It woudl be better to instantiate a new config and set it in the container. (and the other places that use the config file should be removed later)